src/lib/PageTracker.js

```diff
@@ -0,0 +1,410 @@
+     * [constructor]
+     * @param {Object} options - Content Insights Related Options
+     * @param {Object} file - File that is being previewed
+     * @return {PageTracker} PageTracker instance
+     */
+    constructor(options = {}, file) {
+        super();
+        this.setOptions(options);
+        this.file = file;
+        this.activeUserHandler = this.activeUserHandler.bind(this);
+        this.inactiveUserHandler = this.inactiveUserHandler.bind(this);
+        this.throttledActiveUserHandler = throttle(this.activeUserHandler, ACTIVE_USER_THROTTLE, { trailing: false });
+    }
+
+    /**
+     * [destructor]
+     *
+     * @return {void}
+     */
+    destroy() {
+        this.stopTracking();
+        this.unbindIdleUserEvents();
+        this.clearIdleTimeout();
+        this.currentPage = null;
+        this.file = {};
+        this.sessionId = null;
+        this.startTime = null;
+    }
+
+    /**
+     * Get the session id
+     *
+     * @public
+     * @return {string}
+     */
+    getSessionId() {
+        return this.sessionId;
+    }
+
+    /**
+     * Call the interval starting function and set the idle timer
+     *
+     * @public
+     * @return {void}
+     */
+    init() {
+        // Generate a new session id. This would need to be later changed to be generated on server side
+        if (!this.sessionId) {
+            this.sessionId = uuidv4();
+        }
+
+        if (!this.userEventsBounded) {
+            this.bindIdleUserEvents();
+        }
+        // Start tracking time on file if content insights is turn on for that particular file
+        this.setIdleTimer();
+        this.startTracking();
+    }
+
+    /**
+     * Wheter Advanced Content Insights is on/off
+     *
+     * @public
+     * @return {boolean}
+     */
+    isActive() {
+        return this.active;
+    }
+
+    /**
+     * Start the content insights interval, to report data each number of seconds defined on a Const
+     *
+     * @public
+     * @return {void}
+     */
+    startTracking() {
+        this.stopTracking();
+        this.startTime = Date.now();
+
+        this.reportingIntervalId = setInterval(() => {
+            this.report();
+        }, REPORTING_INTERVAL);
+    }
+
+    /**
+     * Stop the content insights interval, for example, if a user goes idle or the user changes the page
+     * and report the data
+     *
+     * @public
+     * @return {void}
+     */
+    stopTracking() {
+        if (!this.reportingIntervalId) return;
+
+        clearInterval(this.reportingIntervalId);
+        this.reportingIntervalId = null;
+        this.report();
+    }
+
+    /**
+     * Handle the page change event. At this moment, it would stop the current timer interval,
+     * report the data for the previous page, and start the interval again
+     *
+     * @public
+     * @param {Object} eventData - Event data payload received by a Page change event
+     * @return {void}
+     */
+    handleViewerPageChange(eventData) {
```

Comment:
We're duplicating a fair bit of state by tracking the current and total pages here as well as the viewer. Can we pass in those values from the viewer event handlers, instead?